### PR TITLE
Tests - Fix failures in tests/unit_tests/transformer/test_attention.py

### PR DESCRIPTION
**Pull request by yzygitzh**
Fix failures in `tests/unit_tests/transformer/test_attention.py` caused by new attention mask shape `(mbs, 1, 1, seq)` required by latest interface of TransformerEngine, from both [NVIDIA]( https://github.com/NVIDIA/TransformerEngine/blob/v2.4/transformer_engine/pytorch/attention/dot_product_attention/utils.py#L617-L634) and [AMD](https://github.com/ROCm/TransformerEngine/blob/release_v1.14_rocm/transformer_engine/pytorch/attention.py#L641-L658).

### CODE REVIEW

**Review by cp5555 — APPROVED** [review]

**Review by abuccts — APPROVED** [review]

**Comment by yzygitzh** [comment]
Close since root cause is found and the failed case is masked flaky.
